tests/test_lint_maturity.py

```diff
@@ -0,0 +1,300 @@
+    "exception": [("package_EX2", "testing/2020.3/rc1")],
+}
+
+RELEASE_SAMPLE_YML = """package_a1: v3.1.a1
+package_b1: v3.1.b1
+package_rc1: v3.1.rc1
+package_ST1: v0.10.4
+package_IV1: 5.13.1-src
+package_EX2: testing/2020.3/rc1"""
+
+TAG_EXCEPTIONS = {"release": ["bleeding", "rpath"], "package": ["package_EX2"]}
+
+TAG_EXCEPTIONS_YML = """release: ['bleeding', 'rpath']
+package: ['package_EX2']"""
+
+RELEASE_FILE_NAMES = [
+    "2020.02.01-py27.yml",
+    "2020.02.a1-py27.yml",
+    "2020.02.b1-py27.yml",
+    "2020.02.rc1-py27.yml",
+    "bleeding-py27.yml",
+]
+
+RELEASE_VERSION_STRINGS = [
+    "2020.02.01",
+    "2020.02.a1",
+    "2020.02.b1",
+    "2020.02.rc1",
+    "bleeding",
+]
+
+EXPECTED_RELEASE_TYPE = [
+    "stable",
+    "a",
+    "b",
+    "rc",
+    "invalid",
+]
+
+EXPECTED_RELEASE_VERSION = [
+    "stable",
+    "a",
+    "b",
+    "rc",
+    "exception",
+]
+
+INVALID_TAGS = {
+    "a": ["invalid"],
+    "b": ["a", "invalid"],
+    "rc": ["a", "b", "invalid"],
+    "stable": ["a", "b", "rc", "invalid"],
+    "exception": ["a", "b", "rc", "invalid"],
+}
+
+EXPECTED_N_INVALID_TAGS = {
+    "a": 1,
+    "b": 2,
+    "rc": 3,
+    "stable": 4,
+    "exception": 4,
+}
+
+EXPECTED_SYSTEMEXIT = """2020.02.01-py27.yml has 4 packages with invalid maturity tag.
+\tTag a packages: [('package_a1', 'v3.1.a1')]
+\tTag b packages: [('package_b1', 'v3.1.b1')]
+\tTag rc packages: [('package_rc1', 'v3.1.rc1')]
+\tTag invalid packages: [('package_IV1', '5.13.1-src')]
+
+2020.02.a1-py27.yml has 1 packages with invalid maturity tag.
+\tTag invalid packages: [('package_IV1', '5.13.1-src')]
+
+2020.02.b1-py27.yml has 2 packages with invalid maturity tag.
+\tTag a packages: [('package_a1', 'v3.1.a1')]
+\tTag invalid packages: [('package_IV1', '5.13.1-src')]
+
+2020.02.rc1-py27.yml has 3 packages with invalid maturity tag.
+\tTag a packages: [('package_a1', 'v3.1.a1')]
+\tTag b packages: [('package_b1', 'v3.1.b1')]
+\tTag invalid packages: [('package_IV1', '5.13.1-src')]
+
+"""
+
+EXPECTED_WARNING = """2020.02.01-py27.yml, exception list of packages:
+\t[('package_EX2', 'testing/2020.3/rc1')] 
+2020.02.a1-py27.yml, exception list of packages:
+\t[('package_EX2', 'testing/2020.3/rc1')] 
+2020.02.b1-py27.yml, exception list of packages:
+\t[('package_EX2', 'testing/2020.3/rc1')] 
+2020.02.rc1-py27.yml, exception list of packages:
+\t[('package_EX2', 'testing/2020.3/rc1')] 
+bleeding-py27.yml not lint because it is in the exception list.
+bleeding-py27.yml, exception list of packages:
+\t[('package_EX2', 'testing/2020.3/rc1')] 
+bleeding-py27.yml has 4 packages with invalid maturity tag.
+\tTag a packages: [('package_a1', 'v3.1.a1')]
+\tTag b packages: [('package_b1', 'v3.1.b1')]
+\tTag rc packages: [('package_rc1', 'v3.1.rc1')]
+\tTag invalid packages: [('package_IV1', '5.13.1-src')]
+"""
+
+
+def _create_tmp_test_files(release_sample, file_names_sample):
+
+    folder_name = os.getcwd() + "/test_lint_maturity/"
```

Comment:
Best to use `os.path.join()` when creating paths, that way it is platform agnostic.